contracts/smart-account/modules/AccountRecoveryModule.sol

```diff
@@ -0,0 +1,580 @@
+        TimeFrame timeFrame
+    );
+    event ThresholdChanged(address indexed smartAccount, uint8 threshold);
+    event SecurityDelayChanged(
+        address indexed smartAccount,
+        uint48 securityDelay
+    );
+
+    error AlreadyInitedForSmartAccount(address smartAccount);
+    error ZeroGuardian();
+    error InvalidTimeFrame(uint48 validUntil, uint48 validAfter);
+    error ExpiredValidUntil(uint48 validUntil);
+    error GuardianAlreadySet(bytes32 guardian, address smartAccount);
+    error GuardianNotSet(bytes32 guardian, address smartAccount);
+    error ThresholdTooHigh(uint8 threshold, uint256 guardiansExist);
+    error ZeroThreshold();
+    error InvalidAmountOfGuardianParams();
+    error GuardiansAreIdentical();
+    error GuardianNotExpired(bytes32 guardian, address smartAccount);
+    error EmptyRecoveryCallData();
+    error RecoveryRequestAlreadyExists(
+        address smartAccount,
+        bytes32 requestCallDataHash
+    );
+
+    /**
+     * @dev Initializes the module for a Smart Account.
+     * Can only be used at a time of first enabling the module for a Smart Account.
+     * @param guardians the list of guardians
+     * @param timeFrames validity timeframes for guardians
+     * @param recoveryThreshold how many guardians' signatures are required to authorize recovery request
+     * @param securityDelay amount of time required to pass between the submission of the recovery request
+     * and its execution
+     * @dev no need for explicit check `length == 0` as it is covered by `recoveryThreshold > length` and
+     * `recoveryThreshold == 0` cheks. So length can never be 0 while recoveryThreshold is not 0
+     */
+    function initForSmartAccount(
+        bytes32[] memory guardians,
+        TimeFrame[] memory timeFrames,
+        uint8 recoveryThreshold,
+        uint48 securityDelay
+    ) external returns (address) {
+        uint256 length = guardians.length;
+        if (_smartAccountSettings[msg.sender].guardiansCount > 0)
+            revert AlreadyInitedForSmartAccount(msg.sender);
+        if (recoveryThreshold > length)
+            revert ThresholdTooHigh(recoveryThreshold, length);
+        if (recoveryThreshold == 0) revert ZeroThreshold();
+        if (length != timeFrames.length) revert InvalidAmountOfGuardianParams();
+        _smartAccountSettings[msg.sender] = SaSettings(
+            uint8(length),
+            recoveryThreshold,
+            securityDelay
+        );
+        for (uint256 i; i < length; ) {
+            if (guardians[i] == bytes32(0)) revert ZeroGuardian();
+            if (timeFrames[i].validUntil == 0)
+                timeFrames[i].validUntil = type(uint48).max;
+            if (timeFrames[i].validUntil < timeFrames[i].validAfter)
+                revert InvalidTimeFrame(
+                    timeFrames[i].validUntil,
+                    timeFrames[i].validAfter
+                );
+            if (
+                timeFrames[i].validUntil != 0 &&
```

Comment:
`timeFrames[i].validUntil != 0` might have a redundant condition since `timeFrames[i].validUntil` is set to `type(uint48).max`

Reply:
Thank you! 🙏